lib/clAdapters/google/oauthCalendar/index.ts

```diff
@@ -0,0 +1,292 @@
+    if (err) {
+      let mapped = err;
+      if (err instanceof Error) {
+        // Map to custom errors
+        if (/unauthorized_client/.test(err.message.toString())) {
+          mapped = createGoogleError(
+            'UnauthorizedClient',
+            err
+          );
+        }
+        // TODO: other types
+      } else if (!err.kind) {
+        // Not a GoogleError
+        mapped = new Error(JSON.stringify(err));
+      }
+      // Leave GoogleErrors
+      rej(mapped);
+    } else {
+      res(typeof returnVal !== 'undefined' ?  returnVal : result);
+    }
+  };
+}
+
+
+
+export const fieldNameMap = {
+  // Desired...                          // Given...
+  'eventId':                             'id',
+  'attendees':                           'attendees',
+  'dateTimeCreated':                     'created',
+  'dateTimeLastModified':                'updated',
+  'attendeeAddress':                     'EmailAddress.Address',
+  'attendeeName':                        'EmailAddress.Name',
+  'iCalUId':                             'iCalUID',
+  'location':                            'location',
+  'status':                              'status',
+  'isCreator':                           'creator.self',
+  'isOrganizer':                         'organizer.self',
+  'organizerEmail':                      'organizer.email',
+  'recurrence':                          'recurrence',
+  'responseStatus':                      'responseStatus',
+  'seriesMasterId':                      'recurringEventId',
+  'dateTimeStart':                       'start.dateTime',
+  'dateTimeEnd':                         'end.dateTime',
+  'subject':                             'summary',
+  'url':                                 'htmlLink',
+  'hangoutLink':                         'hangoutLink',
+  'privacy':                             'visibility'
+};
+
+
+
+export interface UserProfile {
+  email: string;
+  emailAfterMapping: string;
+}
+
+
+
+
+export type GoogleCalendarApiEvent = {
+  [K in keyof (typeof fieldNameMap)]?: (typeof fieldNameMap)[K];
+};
+
+export interface GoogleCalendarApiResult {
+  items: GoogleCalendarApiEvent[];
+  nextPageToken?: string;
+}
+
+
+
+export default class GoogleOauthCalendarAdapter extends GoogleBaseAdapter {
+
+  static Configuration = Configuration;
+  static Service = Service;
+
+  // convert the names of the api response data
+  static fieldNameMap = fieldNameMap;
+
+
+
+  _config: Configuration;
+  _service: Service;
+
+  // constructor needs to call super
+  constructor() {
+    super();
+  }
+
+
+  reset() {
+    delete this._config;
+    delete this._service;
+    return this;
+  }
+
+
+  async init() {
+    const { credentials }: { credentials: {[k: string]: string} } = this;
+
+    if (!credentials) {
+      throw new Error('credentials required for adapter.');
+    }
+
+
+    // validate required credential properties
+    ['access_token', 'refresh_token', 'email'].forEach(prop => {
```

Comment:
Ahh yea, I was going to ask how to get rid of that "can't decrypt/encrypt adapter" message, I guess this will solve that issue.